models/baseModels/Invoice/Invoice.ts

```diff
@@ -752,6 +785,37 @@ export abstract class Invoice extends Transactional {
     });
   }
 
+  async updateIsItemsFullyReturned(doc?: Invoice) {
+    let mainDoc;
+    if (doc?.returnAgainst) {
+      mainDoc = await this.fyo.doc.getDoc(
+        ModelNameEnum.SalesInvoice,
+        doc.returnAgainst
+      );
+    }
+
+    let invoiceDocs: Invoice;
+    if (doc) {
+      invoiceDocs = mainDoc as Invoice;
+    } else {
+      invoiceDocs = this as Invoice;
+    }
+
+    const sumOfReturnDocs = await getReturnQtyTotal(invoiceDocs);
```

Comment:
Rename mainDoc to something clearer
```suggestion
      let sinvDoc;

    if (doc?.returnAgainst) {
      sinvDoc = await this.fyo.doc.getDoc(
        ModelNameEnum.SalesInvoice,
        doc.returnAgainst
      ) as Invoice;
    }

    const totalQtyOfReturnedItems = await getReturnQtyTotal(sinvDoc ?? this);
```